src/main/java/com/websafe/ccmsbe/service/PrivacyRegulationService.java

```diff
@@ -0,0 +1,162 @@
+    public static class WebsiteNotFoundException extends RuntimeException {
+        public WebsiteNotFoundException(String message) {
+            super(message);
+        }
+    }
+
+    //User define exception for PrivacyRegulationUpdate errors
+    public static class PrivacyRegulationUpdateException extends RuntimeException {
+        public PrivacyRegulationUpdateException(String message) {
+            super(message);
+        }
+    }
+
+    //User define exception for PrivacyRegulationCreation errors
+    public static class PrivacyRegulationCreationException extends RuntimeException {
+        public PrivacyRegulationCreationException(String message) {
+            super(message);
+        }
+    }
+
+
+    //User define exception for PrivacyRegulationRetrieval errors
+    public static class PrivacyRegulationRetrievalException extends RuntimeException {
+        public PrivacyRegulationRetrievalException(String message) {
+            super(message);
+        }
+    }
+
+    @Autowired
+    public PrivacyRegulationService(PrivacyRegulationRepository privacyRegulationRepository, WebsiteRepository websiteRepository) {
+        this.privacyRegulationRepository = privacyRegulationRepository;
+        this.websiteRepository = websiteRepository;
+    }
+
+    public PrivacyRegulation createPrivacyRegulation(PrivacyRegulation privacyRegulation) {
+        try {
+            if (privacyRegulation == null) {
+                throw new IllegalArgumentException("Privacy regulation cannot be null");
+            }
+
+            return privacyRegulationRepository.save(privacyRegulation);
+        } catch (IllegalArgumentException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new PrivacyRegulationCreationException("An error occurred while creating the privacy regulation");
+        }
+    }
+
+    public List<PrivacyRegulation> getAllPrivacyRegulations() {
+        try {
+            return privacyRegulationRepository.findAll();
+        } catch (Exception e) {
+            throw new PrivacyRegulationRetrievalException("An error occurred while retrieving all privacy regulations");
+        }
+    }
+
+    @Transactional
+    public String addPrivacyRegulationToWebsite(Long websiteId, Long regulationId) {
+        try {
+            Website website = websiteRepository.findById(websiteId).orElse(null);
+
+            PrivacyRegulation privacyRegulation = privacyRegulationRepository.findById(regulationId).orElse(null);
+
+            if (website == null || privacyRegulation == null) {
+                throw new PrivacyRegulationNotFoundException("Invalid website ID or regulation ID");
+            }
+
+            if (website.getPrivacyRegulations().contains(privacyRegulation)) {
+                throw new PrivacyRegulationNotFoundException("Regulation already added to the website");
+            }
+
+            website.getPrivacyRegulations().add(privacyRegulation);
+
+            websiteRepository.save(website);
+
```

Comment:
Better if we can implement a logger for this code base. It will provide more flexibility in debugging.